src/components/MainPage.js

```diff
@@ -0,0 +1,188 @@
+import * as React from "react";
+import AppBar from "@mui/material/AppBar";
+import Button from "@mui/material/Button";
+import CameraIcon from "@mui/icons-material/PhotoCamera";
+import Card from "@mui/material/Card";
+import CardActions from "@mui/material/CardActions";
+import CardContent from "@mui/material/CardContent";
+import CardMedia from "@mui/material/CardMedia";
+import CssBaseline from "@mui/material/CssBaseline";
+import Grid from "@mui/material/Grid";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import Container from "@mui/material/Container";
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import Labels from "./Labels";
+import { Chip } from "@mui/material";
+import Progress from "./Progress";
+import reactJsLogo from "../logoImages/react-logo.png";
+import jsLogo from "../logoImages/js.svg";
+import dataStructureLogo from "../logoImages/data-structure.png";
+
+const quizCards = [
+  {
+    id: "react-basic",
+    title: "React Basics",
+    category: "react",
+    level: "basic",
+    labels: ["frontend"],
+    image: reactJsLogo,
+  },
+  {
+    id: "react-middle",
+    title: "React Mid-Level",
+    category: "react",
+    level: "mid-level",
+    labels: ["frontend"],
+    image: reactJsLogo,
+  },
+  {
+    id: "js-basic",
+    title: "JS Basics",
+    category: "javaScript",
+    level: "basic",
+    labels: ["frontend", "backend"],
+    image: jsLogo,
+  },
+  {
+    id: "js-functions",
+    title: "JS Functions",
+    category: "javaScript",
+    level: "basic",
+    labels: ["frontend", "backend"],
+    image: jsLogo,
+  },
+  {
+    id: "js-middle",
+    title: "JS Mid-Level",
+    category: "javaScript",
+    level: "mid-level",
+    labels: ["frontend", "backend"],
+    image: jsLogo,
+  },
+  {
+    id: "data-structures",
+    title: "Data structures",
+    category: "data structures",
+    level: "middle",
+    labels: ["frontend", "backend"],
+    image: dataStructureLogo,
+  },
+];
+const userProgress = {
```

Comment:
Do you think you could change this data structure from an object to an array like this:
```
const userProgress = [
    { id: "react-basic", attemptsNumber: "", progress: 50 },
    { id: "react-middle", attemptsNumber: "", progress: 10 },
    { id: "js-basic", attemptsNumber: "", progress: 80 },
    { id: "js-functions", attemptsNumber: "", progress: 100 },
];
```

Reply:
Another option is to make the `userId` an object and rely on the keys of the object as the user ID. This way if you need to look up a user by ID it's very simple. For example:

```jsx
const userProgress = {
  react-basic: {attemptsNumber: '', progress: 50},
  react-middle: {attemptsNumber: '', progress: 10},
  ...
}
```

What do y'all think of this? I assume this is just a placeholder until the backend is completed correct?

Reply:
I communicated with the front-end team, and we decided on the data structure used before! Thank you guys.